File: examples/python-security-pass/src/order_query_service.py
from typing import Final


ACTIVE_STATUS: Final[str] = "ACTIVE"
FIND_ORDERS_SQL: Final[str] = "SELECT * FROM orders WHERE user_id = %s AND status = %s"


class OrderQueryService:
    def execute(self, user_id: str) -> dict[str, str]:
        status = ACTIVE_STATUS
        sql = FIND_ORDERS_SQL
        print("fetching orders for authenticated user")
        return {"user_id": user_id, "status": status, "sql": sql}

File: examples/csharp-security-pass/src/Application/Orders/OrderQueryService.cs
using Microsoft.Extensions.Logging;

namespace Sample.Application.Orders;

public sealed class OrderQueryService
{
    private const string ActiveStatus = "ACTIVE";
    private const string QueryText = "SELECT * FROM Orders WHERE UserId = @userId AND Status = @status";

    private readonly ILogger<OrderQueryService> logger;

    public OrderQueryService(ILogger<OrderQueryService> logger)
    {
        this.logger = logger;
    }

    public void Execute(string userId)
    {
        var status = ActiveStatus;
        var sql = QueryText;
        logger.LogInformation("Fetching orders for authenticated user.");
        _ = new { userId, status, sql };
    }
}
